pulser-core/pulser/sampler/samples.py

```diff
@@ -83,37 +76,91 @@ class ChannelSamples:
     amp: np.ndarray
     det: np.ndarray
     phase: np.ndarray
-
-    slots: list[_TimeSlot] = field(default_factory=list)
+    slots: list[_TargetSlot] = field(default_factory=list)
 
     def __post_init__(self) -> None:
         assert len(self.amp) == len(self.det) == len(self.phase)
+        self.duration = len(self.amp)
 
         for t in self.slots:
             assert t.ti < t.tf  # well ordered slots
-        for t1, t2 in _pairwise(self.slots):
+        for t1, t2 in zip(self.slots, self.slots[1:]):
             assert t1.tf <= t2.ti  # no overlaps on a given channel
 
+    def extend_duration(self, new_duration: int) -> ChannelSamples:
+        """Extends the duration of the samples.
```

Comment:
```suggestion
        """Extends the duration of the samples by padding zero samples at the end.
```

Reply:
It's not exactly this, as the `phase` is not padded with zeros, but I can make the docstring more informative